### DISCUSSION TITLE
integration and feed of data with dAISy hat

### DISCUSSION THREAD

**Discussion by gregbartels** [jvde-github/AIS-catcher#42] [Ideas]
Hi there,

I would like to suggest if is possible integrate the  dAISy hat receiver with your program, I want to share here the link of the product, the hat use the serial port, and include the frequencies in the receiver to be used. Not sure if you can integrate this  soon or if there is a way to integrate easily.  I´m using a script to upload AIS data to APRS but if we can integrate this with the AIS catcher will be great!
https://shop.wegmatt.com/products/daisy-hat-ais-receiver

thank you!

**Comment by jvde-github** [comment]
Hi! Actually the recent NMEA decoder included in the program should help with that and almost get us to process the data from the DaisyHat. I think you can print the messages from the dAISy with a comment like:
```
cat /dev/serial0
```
That means that in theory you could use the following AIS-catcher command:
```
AIS-catcher -r txt /dev/serial0 -m 5 -o 5
```
And add all the UDP HTTP channels you want to add as extra parameters instead of the `-o 5`. This is theory, as I implemented this functionality only recently and have not tested it with dAISy. I have one though and will test it out.

How does your script process the data out of the dAISy and do you provide any additional commands to set things like baud rate?

**Comment by gregbartels** [comment]
Hi Jasper!

Thanks for the response, I have one, if you want I can try to test this in
the next weeks with the dAISy hat.

I'm using this script https://github.com/hsiboy/ais_json
<https://github.com/hsiboy/ais_json>   + Ais dispatcher. This is my setup!
you can see my station on APRS.FI from Central Pacific of Costa Rica.

Thanks in advance!
Greg!



El sáb, 26 nov 2022 a las 13:30, Jasper ***@***.***>)
escribió:

> Hi! Actually the recent NMEA decoder included in the program should help
> with that and almost get us to process the data from the DaisyHat. I think
> you can print the messages from the dAISy with a comment like:
>
> cat /dev/serial0
>
> That means that in theory you could use the following AIS-catcher command:
>
> AIS-catcher -r txt /dev/serial0 -m 5 -o 5
>
> And add all the UDP HTTP channels you want to add as extra parameters
> instead of the -o 5. This is theory, as I implemented this functionality
> only recently and have not tested it with dAISy. I have one though and will
> test it out.
>
> How does your script process the data out of the dAISy and do you provide
> any additional commands to set things like baud rate?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/jvde-github/AIS-catcher/discussions/42#discussioncomment-4243872>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ASWRPICSRGUHEXNBUJALNNLWKJQNTANCNFSM6AAAAAASMAR224>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Gregory Badilla Bartels

**Comment by jvde-github** [comment]
Hi Greg, okay, so I was thinking of having AIS-catcher directly reading from the serial port of the dAISy. I will try that tomorrow. Let's call that option 1 :-)

Understand that you use AISdispatcher to read from the dAISy and send that per UDP messages to your script (option 2). Let's assume this UDP address is127.0.0.1 port 5000 (as per your script). You can get this into AIS-catcher as follows:
```
netcat -l -u 127.0.0.1 5000 | AIS-catcher -r txt . -o 5
```
And then again replace `-o 5` with whatever you want, e.g. use `-H` to get it to APRS.fi.

Keep me posted, I will let you know how my experiment with the dAISy goes. 

Thanks!

**Comment by jvde-github** [comment]
UPDATED after some changes to the program

I have tested reading directly from the dAISY and the following did work for me (with the latest development version of AIS-catcher). First set the baud rate:
```
stty -F /dev/serial0 38400
```
And then you can use AIS-catcher as feeder as follows (in this case sending the data to UDP):
```
AIS-catcher -r txt /dev/serial0 -n -v 60 -u 192.168.1.235 10110
```
Seems to be working smoothly with two small fixes required some day (terminating the program with CTRL-C does not work smoothly and <del>option to select `-m 5` in the config file</del>).

The corresponding config file:
```json
{"config":"aiscatcher","version":1,"input":"rawfile","verbose":true,"file":{"file":"/dev/serial0","format":"txt"}}
``` 

You can add all APRS and UDP channels to the above like normal.
I did have the problem that the serial port disappeared at some point and reading from ``/dev/serial0`` could only be done as root and kept terminating. I don't regularly run the dAISy so interested in your experience.

Keep me posted.
Thanks, Jasper

**Comment by gregbartels** [comment]
Hi Jasper,

Thank you for the confirmation, I will work to have setup this and test
with the AIS-catcher. Thanks for the information shared!!!

Regards from Costa Rica!

El dom, 27 nov 2022 a las 5:05, Jasper ***@***.***>)
escribió:

> I have tested reading directly from the dAISY and the following did work
> for me (with the latest development version of AIS-catcher). First set the
> baud rate:
>
> stty -F /dev/serial0 38400
>
> And then you can use AIS-catcher as feeder as follows (in this case
> sending the data to UDP):
>
> AIS-catcher -r txt /dev/serial0 -m 5 -o 5 -v 60 -u 192.168.1.235 10110
>
> Seems to be working smoothly with two small fixes required some day
> (terminating the program with CTRL-C does not work smoothly and option to
> select -m 5 in the config file).
>
> Keep me posted.
> Thanks, Jasper
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/jvde-github/AIS-catcher/discussions/42#discussioncomment-4246214>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ASWRPIA7MTUBSEMEJEB3EZ3WKM56DANCNFSM6AAAAAASMAR224>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Gregory Badilla Bartels

**Comment by gregbartels** [comment]
Hi Jasper,

Happy New Year 2023!  I follow the steps you provide me but I'm still
unable to see the AIS-Catcher uploading the data.   I'm getting this error:

[image: image.png]


if i run this command   AIS-catcher -r txt /dev/serial0 -m 5 -o 5 -v 60 -u
127.0.0.1 5000 pointing to the port 5000 ( which is my script uploading the
info)   I can see the tool working

[image: image.png]


 but is not reflecting the data in the map of APRS.

[image: image.png]



I will appreciate your help, I apologize to bother you, I'm still learning
about how to configure this, I followed up the steps you shared with me but
I'm still unable to see it the data.

Thanks in advance!

Greg
TI2GBB



El lun, 28 nov 2022 a las 17:51, Gregory Badilla ***@***.***>)
escribió:

> Hi Jasper,
>
> Thank you for the confirmation, I will work to have setup this and test
> with the AIS-catcher. Thanks for the information shared!!!
>
> Regards from Costa Rica!
>
> El dom, 27 nov 2022 a las 5:05, Jasper ***@***.***>)
> escribió:
>
>> I have tested reading directly from the dAISY and the following did work
>> for me (with the latest development version of AIS-catcher). First set the
>> baud rate:
>>
>> stty -F /dev/serial0 38400
>>
>> And then you can use AIS-catcher as feeder as follows (in this case
>> sending the data to UDP):
>>
>> AIS-catcher -r txt /dev/serial0 -m 5 -o 5 -v 60 -u 192.168.1.235 10110
>>
>> Seems to be working smoothly with two small fixes required some day
>> (terminating the program with CTRL-C does not work smoothly and option to
>> select -m 5 in the config file).
>>
>> Keep me posted.
>> Thanks, Jasper
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/jvde-github/AIS-catcher/discussions/42#discussioncomment-4246214>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/ASWRPIA7MTUBSEMEJEB3EZ3WKM56DANCNFSM6AAAAAASMAR224>
>> .
>> You are receiving this because you authored the thread.Message ID:
>> ***@***.***>
>>
>
>
> --
> Gregory Badilla Bartels
>
>

-- 
Gregory Badilla Bartels

**Comment by jvde-github** [comment]
Hi Greg, I cannot see the pictures with the error message. Can you try to reshare?

**Reply by gregbartels** [reply]
No problem!

Let me paste here the information:




_Happy New Year 2023!  I follow the steps you provide me but I'm still unable to see the AIS-Catcher uploading the data.   I'm getting this error:_ 

![image](https://user-images.githubusercontent.com/78452640/216064497-3baad50a-0554-4c0b-b2ec-283c5706abe8.png)


_if i run this command   AIS-catcher -r txt /dev/serial0 -m 5 -o 5 -v 60 -u 127.0.0.1 5000        pointing to the port 5000 ( which is my script uploading the info)   I can see the tool working_

![image](https://user-images.githubusercontent.com/78452640/216064559-182b289b-653f-4ed5-8ab7-167497f96cc2.png)


 _but is not reflecting the data in the map of APRS._

![image](https://user-images.githubusercontent.com/78452640/216064626-0000a071-7efb-4d8f-96f7-f01e76565cd3.png)



_I will appreciate your help, I apologize to bother you, I'm still learning about how to configure this, I followed up the steps you shared with me but I'm still unable to see it the data.

Thanks in advance!

Greg
TI2GBB_

**Comment by jvde-github** [comment]
Hi greg,

The first step seems to be working, which is to get the NMEA lines from your device in AIS-catcher:
```console
AIS-catcher -r txt /dev/serial0 -m 5 -o 5 -v 60 
```
You send this to a UDP address and then pick it up again with another instance with your netcat call. Which is totally fine. But of course you can send to APRS directly from the first instance by using the H switch.

The problem is in your use of the `-H` option which needs more information to be able to send it to APRS. The README of the program has an example. Essentially the `-H` needs to be followed by the following:
```
AIS-catcher -H http://aprs.fi/jsonais/post/secret-key ID callsign PROTOCOL aprs INTERVAL 30
```
The `secret-key` is the key you received from APRS to send your AIS data to and callsign your station name. I think I followed these instructions to get to that: https://aprs.fi/page/ais_feeding

So to read 1) from your device, 2) send to APRS and 3) send to UDP port 5000, you can use one long statement:
```console
AIS-catcher -r txt /dev/serial0 -m 5 -o 5 -v 60  -H http://aprs.fi/jsonais/post/secret-key ID callsign PROTOCOL aprs INTERVAL 30 -u 127.0.0.1 5000
```
I think that is the issue I see (i.e no parameters after `-H`) with your statement but please let me know if I misunderstood.

Instead of the map I am using this [site](https://aprs.fi/page/ais_sites) to check whether there is any data received. You might also see an error message in the AIS-catcher output with the response from the APRS site. But instead of `-o 5` you could consider to use `-q` to suppress all the NMEA lines so that is more visible. 

Thanks for asking, your question has actually resulted in developing this functionality so much appreciated.

Jasper

**Reply by gregbartels** [reply]
Thanks for the extensive explanation  Jasper!


I tried the first  command and I got this:

![image](https://user-images.githubusercontent.com/78452640/216337826-96ece3f4-f4fa-428a-bc5c-681040b8a0c9.png)



I tried the second option ( using AIS Dispatcher) and I got the same:

![image](https://user-images.githubusercontent.com/78452640/216338153-54310ed8-4a29-4681-b2df-9c093463318c.png)


the AIS receiver is called YS1RM  


If I can simplify this using all the ais-catcher software instead of AIS dispatcher it will be great, my idea is experiment and  use all the resources you have developed on the project.  

Again, thanks a lot for the explanations, I will create a guide to use dAISy hat with Ais-catcher as soon as we get this work a expected.

Greg

**Comment by jvde-github** [comment]
Thank you Greg. The issue is that your AIS-catcher build does not support APRS submission as it was  not build with the CURL library included. 
If you type:
```
AIS-catcher -L
```
You need to see something like:
```
Other support: CURL ZLIB RTLSDR-BIASTEE RTLSDR-TUNERBW 
```
which includes CURL and ZLIB. 
I think you have to rebuild AIS-cacther from scratch but before that install the necessary libraries:
```
sudo apt install libcurl4-openssl-dev zlib1g-dev
```
If you in that process run cmake, you should see something like:
```
-- Checking for module 'libcurl'
--   Found libcurl, version 7.74.0
-- CURL: found - /usr/include/aarch64-linux-gnu, /usr/lib/aarch64-linux-gnu/libcurl.so
-- Checking for module 'zlib'
--   Found zlib, version 1.2.11
-- ZLIB: found - /usr/include, /usr/lib/aarch64-linux-gnu/libz.so
```
which tells you that the system could find CURL and ZLIB which are needed for APRS submissions.

Regards, Jasper

**Reply by gregbartels** [reply]
Ok,  I found this on the Pi:

![image](https://user-images.githubusercontent.com/78452640/216405098-d1d54fd5-e742-470c-8047-e93e39f63238.png)


It's curious because Curl is already installed:

![image](https://user-images.githubusercontent.com/78452640/216405375-9fb8e88f-bd26-4976-a499-26b3148fe7f1.png)

**Comment by jvde-github** [comment]
Hi Greg, 

curious that the build of the program could not find the curl library then. 

What do you get with the commands:
```
pkg-config --libs libcurl
```
and
```
pkg-config --cflags libcurl
```
And if you build the latest version of the program what do you see in the cmake step on curl?
You should see something like:
```
-- Checking for module 'libcurl'
--   Found libcurl, version 7.74.0
-- CURL: found - /usr/include/aarch64-linux-gnu, /usr/lib/aarch64-linux-gnu/libcurl.so
-- Checking for module 'zlib'
--   Found zlib, version 1.2.11
-- ZLIB: found - /usr/include, /usr/lib/aarch64-linux-gnu/libz.so
```

**Reply by gregbartels** [reply]
Jasper if I ran the first command I got this:

![image](https://user-images.githubusercontent.com/78452640/216418076-13b80a96-ddd9-4abe-9d9a-d1aa4bc4115f.png)


for the second command


![image](https://user-images.githubusercontent.com/78452640/216418314-51ef9d90-6f1b-4259-a5df-c6f9fa654019.png)




My only doubt is this section ( I apologize, I'm still learning to use linux) 

_And if you build the latest version of the program what do you see in the cmake step on curl?
You should see something like:

-- Checking for module 'libcurl'
--   Found libcurl, version 7.74.0
-- CURL: found - /usr/include/aarch64-linux-gnu, /usr/lib/aarch64-linux-gnu/libcurl.so
-- Checking for module 'zlib'
--   Found zlib, version 1.2.11
-- ZLIB: found - /usr/include, /usr/lib/aarch64-linux-gnu/libz.so_

Should I use cmake libcurl    in the terminal console to get this output or how is the sintax?  Thank you! sorry again for the dummies question.



If I need to start from scratch,  I just need to run remove command to delete the AIS-catcher file and that's all, right?

**Comment by jvde-github** [comment]
No worries. 

By the way the first outcome looks promising. 

Apologies. You need to follow the steps below the table in the instruction: https://github.com/jvde-github/AIS-catcher#ubuntu-raspberry-pi-macos-msvc

```
git clone https://github.com/jvde-github/AIS-catcher.git
cd AIS-catcher
mkdir build
cd build
cmake ..
make
sudo make install
```

So you need to enter these command is this order (in some directory where you want the program and if you already have an AIS-catcher directory you might have to delete it first).
You see that one of the commands is `cmake ..` that should as part of the output show the lines above. 

Feel free to post the output of these steps here (or just send me an email)

**Reply by gregbartels** [reply]
Good day Jasper!


I did the installation again, I follow the steps to setup the port for the dAISy hat,   and I run the next command to  feed directly to   APRS.FI





it's been  24 hours now and  finally we have the results!!!

![image](https://user-images.githubusercontent.com/78452640/216626003-5455e408-ca45-4254-b61c-2de807c6cae6.png)



THANK YOU SO MUCH! You don't know how much I appreciate your help on this, I'm very happy for the results! ( I'm considering record a youtube video in spanish/english to explain how to do it this with dAISy hat!)


At think at this point I can continue experiment the next features of the AIS-Catcher. I wonder to know how often the data in APRS.FI is send, every time the boats move or how frequently is the report of positions sent?

Thanks a lot! I'm working to have setup the second AIS-Catcher now. The idea is coverage all central america with AIS receiver.

Can I create a routine to start the AIS catcher feed after  reboot/ cut of electricity ? Just to avoid manual intervention everytime we need to reboot the Pi due updates or stop electricity fluid to the pi.

Thanks again and regards from Costa Rica!
Greg

**Comment by jvde-github** [comment]
Hi Greg, great news! Happy to hear.

The command line argument contained `INTERVAL 30` which essentially tells the program to collect messages for 30 seconds and then send to APRS. For my station I have set it to 60 but you can shorter as well at the expense of more internet traffic. From the APRS page it seems that 60s is quite common.

There is script that someone developed that will install AIS-catcher as a service that automatically starts: https://github.com/abcd567a/install-aiscatcher

And there is a lot of discussion here on that same script: https://discussions.flightaware.com/t/howto-run-ads-b-receiver-ais-receiver-on-same-rpi/85040

If you are comfortable that is an avenue you could try.

**Comment by g7ruh** [comment]
Hi Greg, 
thanks for starting this discussion, most interesting and helped me learn some things.

I now have my AIS-catcher test system feeding aprs.fi

Thanks to Jasper for creating a simple way to feed aprs.fi too, of course.

73 Roger G7RUH



![AIS -APRS-fi feed test 202303081223](https://user-images.githubusercontent.com/5949609/223731392-ea72c078-4c79-4087-b70a-26ff06f5eb45.png)

**Reply by gregbartels** [reply]
Hey you're welcome!  I'm very happy with the Ais Catcher and the different features of this code!  @jvde-github  provides the best  feedback and support! I hope we can continue expanding the usage of this amazing  project!

**Comment by g7ruh** [comment]
Greg,
I, too am very happy with Jasper (and team) and how stuff just gets fixed.

Team Jasper: keep up the good work. I can see AIS-catcher being a one-stop AIS solution, and as Greg says, keep expanding it. I showed it to a visitor yesterday and he was most impresed.

73 de Roger G7RUH

**Comment by jvde-github** [comment]
The latest edge version has now an experimental feature to read directly from the dAISy HAT and other serial devices using the switch `-e`:
```
AIS-catcher -e 38400 /dev/serial0
```

**Reply by R1ch1e-Lee** [reply]
I had to add user aiscat to the dialout group to get this to read from the serial port using abcd567a install script 

sudo adduser aiscat dialout

log off then back on and it worked

**Comment by jerryaquav** [comment]
My old laptop running an even older EasyAIS with a forwarding program I shan't name has died. I would like to run a new Pi4 & dAISy hat set up but can't find any clear instructions. AIS Catcher has a great reputation but I need to find an idiots guide to get it running quickly. The plan is to send data in to the previous sites I was feeding. Can anyone help?

**Reply by Calimerorulez** [reply]
I can't go into detail now, but I've got a similar setup. The dAISy hat provides a serial port if configured correctly (like /dev/serial1). You can then use that port as an input for AIS Catcher.

**Reply by R1ch1e-Lee** [reply]
I have one setup in testing at my home. Here is my config that works with the daisy hat and raspberry pi:

 usr/share/aiscatcher/aiscatcher.conf

 -e 38400 /dev/serial0
 -v 10
 -M DT
 -gr TUNER 38.6 RTLAGC off
 -s 2304k
 -p 3
 -o 4
 -u 127.0.0.1 10110
 -N 8383
 -N PLUGIN_DIR /usr/share/aiscatcher/my-plugins

I haven't tweaked any other parts of the conf file but just enough to get it working

**Reply by jerryaquav** [reply]
If you get a chance, I'd be grateful to learn more.

**Reply by jerryaquav** [reply]
Do you have to do the normal uart_control change for the dAISy hat before loading aiscatcher? I was was receiving ais messages but after installing via the script from abcd567a that was suggested sentences stopped & I couldn't view the web interface. it just returned ERR_CONNECTION_REFUSED. Ran the whole way through and seemed to be successful.

**Reply by R1ch1e-Lee** [reply]
Did you reboot afterwards?  If I remember correctly it didn't start upon installation until a reboot

**Reply by jvde-github** [reply]
Good to reboot indeed and check whether AIS-catcher is actually running:
```
sudo systemctl status aiscatcher.service
```

**Reply by jerryaquav** [reply]
OK I did a clean install of the OS on the Pi4 again and ran the aiscatcher install from abcd567a. Worked and rebooted. Made the change R1ch1e-Lee uses and rebooted again. Checked that the service is running and it is. Hat LEDs flashing and using 'screen' I can see the ais sentences on serial0. But on ip:8383 I still get ERR_CONNECTION_REFUSED! It has to be something small I'm missing.

**Reply by jerryaquav** [reply]
AIS-catcher -l shows 0 devices found

**Reply by jerryaquav** [reply]
Spotted that adding sudo adduser aiscat dialout would help and now we have access.

**Reply by jerryaquav** [reply]
But still 0 devices found

**Reply by jvde-github** [reply]
Using `-l` will only show SDR devices connected to the machine but not serial.

Assuming that your background AIS service is not running (otherwise type `sudo systemctl stop aiscatcher.service`).
Then you can see the NMEA lines in your terminal as follows:
```
jasper@raspberrypi:~ $ stty -F /dev/serial0 38400
jasper@raspberrypi:~ $ cat /dev/serial0
!AIVDM,1,1,,B,13aGth@P00PCph8MjfBP0?w<261h,0*74
```
You can stop this with CTRL-C and instead read these lines in AIS-catcher:
```
jasper@raspberrypi:~ $ AIS-catcher -e 38400 /dev/serial0
AIS-catcher (build Nov 20 2023) v0.54-67-g2f0aa8ab
(C) Copyright 2021-2023 jvde-github and other contributors
This is free software; see the source for copying conditions.There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
!AIVDM,1,1,,A,19NSOph02DPBG=0MhKVsWqEf0@Pq,0*65 ( MSG: 1, REPEAT: 0, MMSI: 636018659)
```

This `-e 38400 /dev/serial0` should be in your aiscatcher.conf file and probably should remove activating any other devices in that file. I hope at least that above two steps help.

You can switch on the background service again with:
```
sudo systemctl start aiscatcher.service
```
And you can check whether it is running (or any errors) with
```
sudo systemctl status aiscatcher.service

**Reply by R1ch1e-Lee** [reply]
what is the output when you type:   netstat -ant

If it shows something like: tcp        0      0 0.0.0.0:8383            0.0.0.0:*               LISTEN

then it is listening for a connection on port 8383

**Reply by jerryaquav** [reply]
Hi Jasper Many thanks for the coaching.  I think I now have it running. It is a lovely interface & once I got used to changing the back end wasn't as daunting as I thought it would be. I will definitely be recommending it to the AIS community.
 
R1ch1e-Lee many thanks for the the additional input to get me going. The whole setup takes up a lot less space that the old laptop and receiver, cables all over the place & power for all.

**Reply by jvde-github** [reply]
Great, happy to hear! Spread the word :-)

**Reply by jerryaquav** [reply]
another addition for your list of stations running AIS-catcher https://www.marinetraffic.com/en/ais/details/stations/10083/_:6fee18c796a96183f56c876a2b26bac6

**Reply by jvde-github** [reply]
I have added it to the Github page list: https://github.com/jvde-github/AIS-catcher/blob/main/README.md#some-stations-with-ais-catcher 

Jasper

**Reply by jvde-github** [reply]
If you want a dashboard on aiscatcher.org. Happy to set up, please send me an email.

**Comment by gregbartels** [comment]
Hello  @jvde-github    Re activating this  discussion group. 

I had to reimage the raspberry pi this morning, when I tried to run the commands to run the Ais Catcher and feed APRS.FI I got this message:

![image](https://github.com/user-attachments/assets/aac61424-d68d-4b88-a24b-62edbfc7835e)

I´m not getting to make this work, Do you know what I´m possible missing here?   Thanks for your assistance!

**Reply by jvde-github** [reply]
Hi Greg, does the file exist? Eg cat /dev/serial0 ?  I think that the serial device might be on another location or not activated via this raspi-config. Have you followed any steps to get the serial set up after installing the new image?

**Reply by gregbartels** [reply]
Hi Jasper,

I just returned to the port this morning and noticed that, after reinstalling the Raspbian OS, my Raspberry Pi is now using /dev/ttyS0 as the serial port to listen the DAISY Hat, I can see its receiving  traffic and packages.

I’m not an expert on this, but I’m wondering if something might be missing or incorrect in the command I’m using to upload the data to APRS and also have it appear in AIS-catcher.

Here’s the command I’m currently using:


_AIS-catcher -r txt /dev/serial0 -m 5 -o 5 -v 60 -H http://aprs.fi/jsonais/post/Kw5qTtf3Qv ID TI2GBB-2 PROTOCOL aprs INTERVAL 30_

Should I be replacing /dev/serial0 with /dev/ttyS0 in this case? Any guidance would be greatly appreciated.

Thanks so much for your help!

Best regards,

**Reply by jvde-github** [reply]
Hi Greg, good to hear from you. And indeed, reinstalling Raspbian OS can change the address of the dAISy Hat. Instead of ` -r txt /dev/serial0` now you need to replace that with `-r txt /dev/ttyS0`.  Did you also specifiy the baud rate somewhere? If so, need to update as well to the new address.

You can also replace ` -r txt /dev/serial0 -m 5` with `-e 38400 /dev/ttyS0`.  This functionality has been added after you started to use the program and directly reads from the serial device, sets the baud rate and picks the right decoding model.

Finally, you are in.a location with not a lot of AIS receivers. Would you be interested in adding -X as well for sharing data with the map aiscatcher.org. Absolutely not needed, but this has been added as well since you are started to use the program.

Does that help?

Kind regards, Jasper

**Reply by gregbartels** [reply]
hi Jasper,

Thank you so much for your response, it was super helpful to  mitigate my issue!


Sure, no problem!   Let me know what will be the final  output to feed the  aiscatcher.org! 


this is the command I'm using

AIS-catcher -r txt /dev/serial0 -m 5 -o 5 -v 60  -H http://aprs.fi/jsonais/post/Kw5qTtf3Qv ID TI2GBB-2 PROTOCOL aprs INTERVAL 30




I have noticed something today,  after some time sending the  data to  APRS.FI  I got this output:


{"class":"AIS","device":"AIS-catcher","version":0,"driver":0,"hardware":"","rxtime":"20250606155034","scaled":true,"channel":"B","nmea":["!AIVDM,1,1,,B,35CqVN1001Iswo:5bebuWDu60Dlb,0*1E"],"signalpower":1024.000000,"ppm":1024.000000,"type":3,"repeat":0,"mmsi":356411000,"status":1,"status_text":"At anchor","turn_unscaled":0,"turn":0,"speed":0.100000,"accuracy":false,"lon":-84.760368,"lat":9.904498,"course":348.500000,"heading":158,"second":35,"maneuver":0,"raim":false,"radio":85290,"sync_state":0,"slot_timeout":5,"received_stations":3370}
{"class":"AIS","device":"AIS-catcher","version":0,"driver":0,"hardware":"","rxtime":"20250606155034","scaled":true,"channel":"B","nmea":["!AIVDM,1,1,,B,3815S6Q000qssmd5cb59858v0Dg:,0*04"],"signalpower":1024.000000,"ppm":1024.000000,"type":3,"repeat":0,"mmsi":538010394,"status":1,"status_text":"At anchor","turn_unscaled":0,"turn":0,"speed":0.000000,"accuracy":true,"lon":-84.774094,"lat":9.930273,"course":233.600006,"heading":164,"second":31,"maneuver":0,"raim":false,"radio":84938,"sync_state":0,"slot_timeout":5,"received_stations":3018}
----------------------
[NMEA input #0-0]                       total: 428 msgs

HTTP Client [http://aprs.fi/jsonais/post/Kw5qTtf3Qv]: return code 200 msg: {"protocol":"jsonais","command":"post","result":"ok","description":"received jsonais POST successfully"}
HTTP: stop thread (http://aprs.fi/jsonais/post/Kw5qTtf3Qv).

After that, the feed stops. Could this indicate that APRS.FI is resetting or terminating the connection?


Thanks in advance!

**Reply by jvde-github** [reply]
That is odd:  "received jsonais POST successfully" generally suggests that sending the data was successful and you should see a similar message every 30 seconds. You could add `RESPONSE ON` to ensure you see the result of sending the data every 30s:
```
AIS-catcher -r txt /dev/serial0 -m 5 -o 5 -v 60 -H http://aprs.fi/jsonais/post/Kw5qTtf3Qv ID TI2GBB-2 PROTOCOL aprs INTERVAL 30 RESPONSE on
```

To share with AIS-catcher, simply add `-X`:
```
AIS-catcher -r txt /dev/serial0 -m 5 -o 5 -v 60 -H http://aprs.fi/jsonais/post/Kw5qTtf3Qv ID TI2GBB-2 PROTOCOL aprs INTERVAL 30 RESPONSE on -X
```

Hope you will see a bit more details now from APRS data uploads to see what is going on

**Reply by gregbartels** [reply]
Thank you for the information!


I got this a few sec ago:



{"class":"AIS","device":"AIS-catcher","version":0,"driver":0,"hardware":"","rxtime":"20250606172325","scaled":true,"channel":"A","nmea":["!AIVDM,1,1,,A,19@>jLO000IsEk:5`109q5:l0@?3,0*24"],"signalpower":1024.000000,"ppm":1024.000000,"type":1,"repeat":0,"mmsi":620999281,"status":15,"status_text":"Not defined","turn_unscaled":0,"turn":0,"speed":0.000000,"accuracy":false,"lon":-84.903938,"lat":9.830827,"course":253.199997,"heading":165,"second":26,"maneuver":0,"raim":false,"radio":66499,"sync_state":0,"slot_timeout":4,"slot_number":963}
{"class":"AIS","device":"AIS-catcher","version":0,"driver":0,"hardware":"","rxtime":"20250606172334","scaled":true,"channel":"A","nmea":["!AIVDM,1,1,,A,3815S6Q000IssuD5cbAbDERv0Dg:,0*4E"],"signalpower":1024.000000,"ppm":1024.000000,"type":3,"repeat":0,"mmsi":538010394,"status":1,"status_text":"At anchor","turn_unscaled":0,"turn":0,"speed":0.000000,"accuracy":false,"lon":-84.773697,"lat":9.930357,"course":264.100006,"heading":177,"second":31,"maneuver":0,"raim":false,"radio":84938,"sync_state":0,"slot_timeout":5,"received_stations":3018}
----------------------
[NMEA input #0-0]                       total: 9 msgs

HTTP Client [http://aprs.fi/jsonais/post/Kw5qTtf3Qv]: return code 200 msg: {"protocol":"jsonais","command":"post","result":"ok","description":"received jsonais POST successfully"}
HTTP: stop thread (http://aprs.fi/jsonais/post/Kw5qTtf3Qv).
TCP (185.77.96.227:4242): disconnected.


It seems for me that APRS.FI is dropping the connection.

**Reply by jvde-github** [reply]
No, it seems that the program stops and then triggers a final upload to APRS.  Assume you are not terminating it?

Can you try to run, using the serial function:

AIS-catcher -e 38400 /dev/serial0  -o 5 -v 60 -H http://aprs.fi/jsonais/post/Kw5qTtf3Qv ID TI2GBB-2 PROTOCOL aprs INTERVAL 30 RESPONSE on -X

And by removing APRS:

AIS-catcher -e 38400 txt /dev/serial0  -o 5 -v 60  -X

And see what happens?

**Reply by gregbartels** [reply]
Ok, I ran the command.

_AIS-catcher -e 38400 /dev/serial0 -o 5 -v 60 -H http://aprs.fi/jsonais/post/Kw5qTtf3Qv ID TI2GBB-2 PROTOCOL aprs INTERVAL 30 RESPONSE on -X_

This was the output after 30min:


HTTP Client [http://aprs.fi/jsonais/post/Kw5qTtf3Qv]: return code 200 msg: {"protocol":"jsonais","command":"post","result":"ok","description":"received jsonais POST successfully"}
{"class":"AIS","device":"AIS-catcher","version":0,"driver":0,"hardware":"","rxtime":"20250606180946","scaled":true,"channel":"A","nmea":["!AIVDM,1,1,,A,19@>jLO000qsEkB5`0wa:E9L0D0G,0*1D"],"signalpower":1024.000000,"ppm":1024.000000,"type":1,"repeat":0,"mmsi":620999281,"status":15,"status_text":"Not defined","turn_unscaled":0,"turn":0,"speed":0.000000,"accuracy":true,"lon":-84.903931,"lat":9.830823,"course":234.500000,"heading":164,"second":46,"maneuver":0,"raim":false,"radio":81943,"sync_state":0,"slot_timeout":5,"received_stations":23}
{"class":"AIS","device":"AIS-catcher","version":0,"driver":0,"hardware":"","rxtime":"20250606180950","scaled":true,"channel":"A","nmea":["!AIVDM,1,1,,A,1815Qs0PApIsp7B5QM4A30iV08Mi,0*2B"],"signalpower":1024.000000,"ppm":1024.000000,"type":1,"repeat":0,"mmsi":538010092,"status":0,"status_text":"Under way using engine","turn_unscaled":-127,"turn":-127,"speed":12.000000,"accuracy":false,"lon":-84.786812,"lat":9.651655,"course":26.800001,"heading":24,"second":51,"maneuver":0,"raim":false,"radio":34673,"sync_state":0,"slot_timeout":2,"slot_number":1905}
Serial read encountered an error: unexpected end.

----------------------
[NMEA input #0-0]                       total: 192 msgs

HTTP Client [http://aprs.fi/jsonais/post/Kw5qTtf3Qv]: return code 200 msg: {"protocol":"jsonais","command":"post","result":"ok","description":"received jsonais POST successfully"}
HTTP: stop thread (http://aprs.fi/jsonais/post/Kw5qTtf3Qv).
TCP (185.77.96.227:4242): disconnected.

![image](https://github.com/user-attachments/assets/213749ae-40ac-450f-ab57-59bffff8d549)




I tried to ran _AIS-catcher -e 38400 txt /dev/serial0 -o 5 -v 60 -X_   and I get this message:


_AIS-catcher -e 38400 txt /dev/serial0 -o 5 -v 60 -X
syntax error on command line with setting "-e". requires two parameters [baudrate] [portname]._

![image](https://github.com/user-attachments/assets/eac1ae84-f09c-4cb4-be58-a389d323f718)

**Reply by jvde-github** [reply]
Yes, the txt in the second line is too much but not needed anymore. We now see what the issue is, the dAISy breaks the connection for some reason. Let me do some research. Is it the same Pi as you used before? Only the Operating System was updates?

If you immediately restart after the stop, does it then run again?

Can you share which Raspberry version you are using?

**Reply by gregbartels** [reply]
Understood!


is the same Raspberry PI I have used before,   only  the operating system was updated.

-Correct, If I restarted after the stop, it will  run again  until the next stop. 


Model		: Raspberry Pi 3 Model B Plus Rev 1.3

**Reply by jvde-github** [reply]
Thank you. I don't think it is an AIS catcher issue but perhaps we can reconnect to the dAISy automatically if it fails. Let me explore and see if I can reproduce.
Thanks Greg

**Reply by jvde-github** [reply]
Question. if you have a new OS you need to go enter
```
sudo rasp-config
```
And follow the below from the dAISy hat manual. Have you done this step and can it resolve the interruption?

![image](https://github.com/user-attachments/assets/6620db3a-eead-46a6-91e6-69aee961da61)

**Reply by gregbartels** [reply]
I applied the steps yesterday... I see a times when the command stopped, however it's been active for almost 9 hours now.

**Reply by jvde-github** [reply]
I discovered that the setting might be such that the output of the serial device is echoed back as input. This caused issues for another serial device. I have updated the code and fixed this setting.
